### PR TITLE
A new feature for the "tls" directive

### PR DESCRIPTION
**Pull request by elcore**
Now it´s possible to do this:

```
domain.com {
    tls me@domain.com {
        key_type ec384 // ec256, rsa8192, rsa4096 or rsa2048
    }
}
```

### CODE REVIEW

File: caddy/https/setup.go [hunk 1]
```diff
@@ -220,6 +228,11 @@ func loadCertsInDir(c *setup.Controller, dir string) error {
 // port to 443 if not already set, TLS is enabled, TLS is manual, and the host
 // does not equal localhost.
 func setDefaultTLSParams(c *server.Config) {
+	// If no KeyType provided, use EC384 as preferred KeyType
+	if KeyType == "" {
+		KeyType = acme.EC384
+	}
```
Comment (mholt) [inline caddy/https/setup.go]: How is the compatibility for EC certs? Maybe we should still default to 2048 bit RSA. What do you think?

**Comment by mholt** [comment]
Nice! :clap: Thanks for working on this. Excited to merge this in.

File: caddy/https/setup.go (hunk 1 above)
Comment (elcore) [inline caddy/https/setup.go]: I will default the cert to 2048 bit RSA now :smile:

File: caddy/https/setup.go [hunk 2]
```diff
@@ -220,6 +228,11 @@ func loadCertsInDir(c *setup.Controller, dir string) error {
 // port to 443 if not already set, TLS is enabled, TLS is manual, and the host
 // does not equal localhost.
 func setDefaultTLSParams(c *server.Config) {
+	// If no KeyType provided, use EC384 as preferred KeyType
```
Comment (mholt) [inline caddy/https/setup.go]: I'm gonna be really nit-picky for just a sec, I know, but let's just delete this comment entirely. It's out of date and doesn't tell us something non-obvious.

Comment (elcore) [inline caddy/https/setup.go]: Pardon @mholt

**Comment by mholt** [comment]
Thanks @elcore! Great work.

**Comment by mholt** [comment]
Also I'm making you a collaborator, so you can help review pull requests and stuff. And if you want to make changes like this or whatever, feel free to open pull requests on a branch in this repo if you want, before committing to master. Cheers!

**Pull request merged**